ext/soap/php_http.c

```diff
@@ -241,6 +241,13 @@ static php_stream* http_connect(zval* this_ptr, php_url *phpurl, int use_ssl, ph
 	if (stream && *use_proxy && use_ssl) {
 		smart_str soap_headers = {0};
 
+		/* Set peer_name or name verification will try to use the proxy server name */
+		if (context && (tmp = php_stream_context_get_option(context, "ssl", "peer_name")) != NULL) {
```

Comment:
I'm glad you caught that. I definitely had it backward. The patch we've been running uses the same logic as http_fopen_wrapper, but in preparing this PR, I pulled the wrong revision from our internal git mirror. I did run the test before submitting, and I've been looking into why it passed when it shouldn't have. The request to download the WSDL file (which uses streams instead of the SOAP HTTP client) was setting peer_name on the context and masking the bug. I've updated the test to avoid this by enabling WSDL caching and warming the cache with a separate SoapClient instance.